docs/DeveloperGuide.md

```diff
@@ -1,38 +1,201 @@
+<!-- omit in toc -->
 # Developer Guide
+<!-- omit in toc -->
```

Comment:
Not specific to the Developer Guide, but couldn't the navigation bar at the top of the page be adjusted? For example, change the "Duke" link to the name of your own program. A link to the User Guide can also improve the experience.